### ISSUE TITLE
Incorrect aliasing.unused AL05 error when aliasing is used inside a subquery

### ISSUE THREAD

**Issue by dineshmarimu2** [sqlfluff/sqlfluff#7039]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a dbt model looking like the one below which on sqlfluff linting produces an aliasing.unused error for the alias t, while it is definitely being used inside a subquery.

### Expected Behaviour

aliasing.unused errors should not be detected

### Observed Behaviour

L:  16 | P:  56 | AL05 | Alias 't' is never used in SELECT statement.
                       | [aliasing.unused]

### How to reproduce

templater = dbt
```
{{ config(schema='stg_bth') }}

SELECT
    team,
    member,
    planfeld AS plan_field,
    TO_DATE((x).key, 'mon_YY') AS month,
    (x).value::float AS plan_value
FROM
    (
        SELECT
            team,
            member,
            planfeld,
            EACH(HSTORE(t)) AS x
        FROM member_plan AS t
    ) AS q
WHERE
    (x).key ~ '^[a-z]{3}_[0-9]{2}$'
    )
```

### Dialect

postgres

### Version

sqlfluff==3.4.2
sqlfluff-templater-dbt==3.4.2

### Configuration
```ini
[sqlfluff]
templater = dbt
sql_file_exts = .sql,.sql.j2,.dml,.ddl
dialect = postgres

# Comma separated list of rules to exclude, or None
# See https://docs.sqlfluff.com/en/stable/perma/rule_disabling.html
# AM04 (ambiguous.column_count) and ST06 (structure.column_order) are
# two of the more controversial rules included to illustrate usage.
exclude_rules =
    ambiguous.column_count,
    ambiguous.column_references,
    ambiguous.distinct,
    layout.long_lines,
    layout.operators,
    references.consistent,
    references.from,
    references.keywords,
    references.qualification,
    references.quoting,
    references.special_chars,
    structure.column_order,
    structure.else_null,
    structure.simple_case,
    structure.subquery,
    structure.using,
    structure.unused_cte,
    structure.unused_join
# aliasing.unused (table aliases not used in the query) is a common rule
# aliasing.self_alias.column (prohibits aliases of form of "a AS a")
# structure.join_condition_order (enforces join conditions to be in a specific order)
# ambiguous.join (ensures that joins are not ambiguous)
# aliasing.expression (enforces that expressions are aliased with AS)

# CPU processes to use while linting.
# The default is "single threaded" to allow easy debugging, but this
# is often undesirable at scale.
# If positive, just implies number of processes.
# If negative or zero, implies number_of_cpus - specified_number.
# e.g. -1 means use all processors but one. 0 means all cpus.
processes = -1

warnings = ST09

[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
template_blocks_indent = False

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

# If using the dbt templater, we recommend setting the project dir.
[sqlfluff:templater:dbt]
project_dir = .
profiles_dir = profile/
profile = dwh-postgres
target = postgres
dbt_skip_compilation_error = True 

# The default configuration for capitalisation rules is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
# Typically we find users rely on syntax highlighting rather than
# capitalisation to distinguish between keywords and identifiers.
# Clearly, if your organisation has already settled on uppercase
# formatting for any of these syntax elements then set them to "upper".
# See https://stackoverflow.com/questions/608196/why-should-i-capitalize-my-sql-keywords-is-there-a-good-reason
[sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = consistent
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = consistent
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = consistent
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = consistent
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = consistent

[sqlfluff:rules:convention.not_equal]
# Consistent usage of preferred "not equal to" comparison
preferred_not_equal_style = consistent
```

.sqlfluffignore
```ini
*target/
# dbt <1.0.0
dbt_modules/
# dbt >=1.0.0
dbt_packages/
dbt_internal_packages/
edr_target/
macros/
.dbt/
.github/
venv/
logs/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
